src/data_provider/qa/test_network.py

```diff
@@ -0,0 +1,57 @@
+import json
+import platform
+import subprocess
+from pathlib import Path
+
+import pytest
+from jsonschema import validate
+from jsonschema.exceptions import ValidationError
+
+
+def call_binary(binary_path):
+    try:
+        # Run the binary and capture its output
+        result = subprocess.run(
+            [binary_path], capture_output=True, check=True, text=True)
+        return result.stdout.strip()
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(f"Error while executing the binary: {e}") from e
+
+
+def validate_network_json(json_data):
+    # Path to the schema file
+    schema_file = Path("qa", "network_schema.json")
+
+    # Load the schema from the file
+    with open(schema_file, "r") as f:
+        network_schema = json.load(f)
+
+    try:
+        # Validate the JSON data against the schema
+        validate(instance=json_data, schema=network_schema)
+    except ValidationError as e:
+        pytest.fail(f"The output does not comply with the schema: {e}")
+
+
+def test_json_output():
+    # Path to the shared library
+    binary_filename = "sysinfo_test_tool.exe" if platform.system(
```

Comment:
Splitting the line in the middle of the ternary expression makes it difficult to read the section.